### PR TITLE
1677: Create End to End BIE Kafka Integration Test

### PR DESCRIPTION
**Pull request by dfitchett**
## What was the problem?

There was no end to end test from mock kafka --> svc-bie-kafka --> xample-workflow --> postgres.

Associated tickets or Slack threads:
- #1677 
- #1680

## How does this fix it?[^1]

Adds continuous integration test. See passing test [here](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/5627186062/job/15249308475).

## How to test this PR
- Run [continuous integration test](https://github.com/department-of-veterans-affairs/abd-vro/actions/workflows/continuous-integration.yml).

## Follow On:
This test should be updated if the `event_details` column in the table `claims.bie_contention_event` is updated. See [V1.1__BIE_Contention_Event.sql](https://github.com/department-of-veterans-affairs/abd-vro/blob/6b6b2ea72a949d7aaf9653319a7024be1eb102d2/db-init/src/main/resources/database/migrations/V1.1__BIE_Contention_Event.sql) and [ContentionEventEntity.java](https://github.com/department-of-veterans-affairs/abd-vro/blob/6b6b2ea72a949d7aaf9653319a7024be1eb102d2/shared/persistence-model/src/main/java/gov/va/vro/persistence/model/ContentionEventEntity.java) which may have been updated by #1680 

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: .github/workflows/bie-kafka-end2end-test-with-mock.yml [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  workflow_call:
+
+concurrency:
+  group: bieKafkaEnd2EndTestWithMock-${{ github.ref }}
+
+env:
+  COMPOSE_PROFILES: 'kafka'
+  TEST_TOPIC: 'TST_CONTENTION_BIE_CONTENTION_ASSOCIATED_TO_CLAIM_V02'
+  # TEST_DATA should be replaced with data after event_details is populated with non PII data. See Github Issue 1680
+  TEST_DATA: 'Lorem ipsum'
```
Comment (dfitchett) [inline .github/workflows/bie-kafka-end2end-test-with-mock.yml:17]: This value is used as the `event_details` column saved in the DB.

**Review by msnwatson — APPROVED** [review]

File: mocks/docker-compose.yml [hunk 2]
```diff
@@ -61,11 +61,15 @@ services:
     <<: [*common-sde-security, *common-security-opt]
     ports:
       - "9092:9092"
+      - "9094:9094"
```
Comment (yoomlam) [inline mocks/docker-compose.yml:64]: Add code comment describing the purpose of exposing this port.

File: .github/workflows/continuous-integration.yml [hunk 3]
```diff
@@ -34,3 +34,7 @@ jobs:
   svc-bgs-api:
     uses: ./.github/workflows/svc-bgs-api-integration-test.yml
     secrets: inherit
+
+  bie-kafka-end-to-end-mock:
```
Comment (yoomlam) [inline .github/workflows/continuous-integration.yml:38]:
```suggestion
  bie-kafka-end-to-end:
```

**Review by yoomlam — CHANGES_REQUESTED** [review]

**Review by yoomlam — APPROVED** [review]
Great! Need to explain the Kakfa internal and external comms a bit more.

File: mocks/docker-compose.yml [hunk 4]
```diff
@@ -59,10 +59,12 @@ services:
     profiles: ["all","kafka"]
     image: va/vro_mocks-mock-bie-kafka:latest
     <<: [*common-sde-security, *common-security-opt]
+    # Port 9092 is for internal communication within the vro_intranet, and is needed for consuming/producing messages
+    # within the vro_intranet docker network. In order to produce/consume messages outside the vro_intranet network,
+    # the port 9094 is exposed for external communication with mock-bie-kafka from outside the vro_intranet. Advertised
+    # listeners are configured to consume messages on both ports.
```
Comment (yoomlam) [inline mocks/docker-compose.yml:65]: Nice!

**Pull request merged**